controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -86,8 +86,12 @@ public void triggerFromCompletion(JobReport report) {
             if (report.success()) {
                 if (report.jobType() == JobType.component) {
                     if (acceptNewApplicationVersionNow(application)) {
-                        // Set this as the change we are doing, unless we are already pushing a platform change
-                        if ( ! ( application.change().platform().isPresent())) {
+                        // If there's an ongoing upgrade, we allow the upgrade and new application package to be
+                        // deployed together
+                        if (application.change().platform().isPresent()) {
+                            application = application.withChange(Change.of(application.change().platform().get(),
+                                                                           applicationVersion));
+                        } else {
```

Comment:
This is the same as 
`application.witChange(application.change().with(applicationVersion)`
so I think you can just replace both the if and else with just that line.

Reply:
Thanks, I'll fix it.